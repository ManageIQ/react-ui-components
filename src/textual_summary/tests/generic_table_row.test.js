import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GenericTableRow from '../generic_table_row';

describe('Simple Table', () => {
  /*
   *  row has
   *    1) label
   *    2) icon
   *    3) value
   *  item.value
   *    a) simple
   *    b) array    TODO
   *  linkClick handler TODO
   */
  it('renders simple value just fine...', () => {
    const item = {
      label: 'Network Manager',
      value: 'Amazon East Network Manager',
      icon: null,
      image: 'http://localhost:3000/assets/svg/vendor-ec2_network-04c432db85be0fd670cd6da83b8685dab51e1b7a63258b70cccbdf8d7f72c988.svg',
      link: '/ems_network/10000000000045',
      title: 'Show Network Manager \'Amazon East Network Manager\'',
      hoverClass: '',
    };

    const row = shallow(<table>
      <tbody>
        <GenericTableRow item={item} onClick={() => null} />
      </tbody>
    </table>);
    expect(toJson(row)).toMatchSnapshot();
  });

  it('renders multi-value just fine...', () => {
    const item = {
      label: 'Network Manager',
      value: [
        { title: 'Amazon East Network Manager', icon: null, value: 'foobar' },
        { title: 'Amazon West Network Manager', icon: null, value: 'boofar' },
      ],
      icon: null,
      image: 'http://localhost:3000/assets/svg/vendor-ec2_network-04c432db85be0fd670cd6da83b8685dab51e1b7a63258b70cccbdf8d7f72c988.svg',
      link: '/ems_network/10000000000045',
      title: 'Show Network Manager \'Amazon East Network Manager\'',
      hoverClass: '',
    };

    const row = shallow(<table>
      <tbody>
        <GenericTableRow item={item} onClick={() => null} />
      </tbody>
    </table>);
    expect(toJson(row)).toMatchSnapshot();
  });
});
