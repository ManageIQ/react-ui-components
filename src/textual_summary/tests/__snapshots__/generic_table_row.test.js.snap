// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Table renders multi-value just fine... 1`] = `
<table>
  <tbody>
    <GenericTableRow
      item={
        Object {
          "hoverClass": "",
          "icon": null,
          "image": "http://localhost:3000/assets/svg/vendor-ec2_network-04c432db85be0fd670cd6da83b8685dab51e1b7a63258b70cccbdf8d7f72c988.svg",
          "label": "Network Manager",
          "link": "/ems_network/10000000000045",
          "title": "Show Network Manager 'Amazon East Network Manager'",
          "value": Array [
            Object {
              "icon": null,
              "title": "Amazon East Network Manager",
              "value": "foobar",
            },
            Object {
              "icon": null,
              "title": "Amazon West Network Manager",
              "value": "boofar",
            },
          ],
        }
      }
      onClick={[Function]}
    />
  </tbody>
</table>
`;

exports[`Simple Table renders simple value just fine... 1`] = `
<table>
  <tbody>
    <GenericTableRow
      item={
        Object {
          "hoverClass": "",
          "icon": null,
          "image": "http://localhost:3000/assets/svg/vendor-ec2_network-04c432db85be0fd670cd6da83b8685dab51e1b7a63258b70cccbdf8d7f72c988.svg",
          "label": "Network Manager",
          "link": "/ems_network/10000000000045",
          "title": "Show Network Manager 'Amazon East Network Manager'",
          "value": "Amazon East Network Manager",
        }
      }
      onClick={[Function]}
    />
  </tbody>
</table>
`;
